sorc/chgres_cube.fd/surface.F90

```diff
@@ -2234,6 +2612,498 @@ subroutine calc_liq_soil_moisture
+ if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN FieldGet", rc)
+
+ print*,"- CALL FieldGet FOR TARGET GRID ALVWF."
+ call ESMF_FieldGet(alvwf_target_grid, &
+                    farrayPtr=alvwf_target_ptr, rc=rc)
+ if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN FieldGet", rc)
+
+ print*,"- CALL FieldGet FOR TARGET GRID ALNWF."
+ call ESMF_FieldGet(alnwf_target_grid, &
+                    farrayPtr=alnwf_target_ptr, rc=rc)
+ if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN FieldGet", rc)
+
+print*,"- CALL FieldGet FOR TARGET GRID FACSF."
+ call ESMF_FieldGet(facsf_target_grid, &
+                    farrayPtr=facsf_target_ptr, rc=rc)
+ if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN FieldGet", rc)
+
+ print*,"- CALL FieldGet FOR TARGET GRID FACWF."
+ call ESMF_FieldGet(facwf_target_grid, &
+                    farrayPtr=facwf_target_ptr, rc=rc)
+ if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN FieldGet", rc)
+
+ print*,"- CALL FieldGet FOR TARGET GRID VEG FRAC."
+ call ESMF_FieldGet(veg_greenness_target_grid, &
+                    farrayPtr=veg_greenness_target_ptr, rc=rc)
+ if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN FieldGet", rc)
+
+ print*,"- CALL FieldGet FOR TARGET GRID MAX VEG FRAC."
+ call ESMF_FieldGet(max_veg_greenness_target_grid, &
+                    farrayPtr=max_veg_greenness_target_ptr, rc=rc)
+ if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN FieldGet", rc)
+
+ print*,"- CALL FieldGet FOR TARGET GRID MIN VEG FRAC."
+ call ESMF_FieldGet(min_veg_greenness_target_grid, &
+                    farrayPtr=min_veg_greenness_target_ptr, rc=rc)
+ if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN FieldGet", rc)
+
+ print*,"- CALL FieldGet FOR TARGET GRID CANOPY MC."
+ call ESMF_FieldGet(canopy_mc_target_grid, &
+                    farrayPtr=canopy_mc_target_ptr, rc=rc)
+ if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN FieldGet", rc)
+
+ print*,"- CALL FieldGet FOR TARGET GRID SNOW ALBEDO."
+ call ESMF_FieldGet(mxsno_albedo_target_grid, &
+                    farrayPtr=mxsno_albedo_target_ptr, rc=rc)
+ if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+    call error_handler("IN FieldGet", rc)
+
+ do i =clb(1),cub(1)
+   do j = clb(2),cub(2)
+     if (landmask_ptr(i,j)==1 .and. soilm_target_ptr(i,j,1) < 0.001) then !.and. &
```

Comment:
What is the cause of these bad points?

Reply:
At GSL's request, we've added in functionality that allows the user to read in vegetation and soil type from the input grid. They wanted the higher resolution data available from HRRR files. This resulted in some mismatched land/water points. Note that all of these subroutines are now only called if the  *_from_climo flags are set to "False", and thus the program reads in these new variables. These new sub routines were required to deal with these problems. You'll find other code in the surface interp subroutine related to these flags as well.

Reply:
Are these fields read in from the 'geogrid' file?  Can I look at this file?  Is it on Hera?

Reply:
They are in the HRRRX files that GSL uses for their HRRR-like runs. @JeffBeck-NOAA should be able to point you to the  correct file.

Reply:
@GeorgeGayno-NOAA, you can find the HRRR geogrid file on Hera here: /scratch2/BMC/det/beck/FV3-LAM/geo_em.d01.nc_HRRRX